clients/slack/client.py

```diff
@@ -96,19 +101,18 @@ def send_report(self, channel_name: str, report_file_path: str):
             logger.error('Failed to send report to Slack.')
         return send_succeed
 
-    def _get_channel_id(self, channel_name: str) -> str:
+    def _get_channel_id(self, channel_name: str) -> Optional[str]:
         try:
             channels = self._get_channels()
-            channel = [
-                channel
-                for channel
-                in channels
-                if channel["name"] == channel_name
-            ][0]
-            return channel["id"]
+            for channel in channels:
+                if channel['name'] == channel_name:
+                    return channel['id']
+            available_channel_names = [channel['name'] for channel in channels]
```

Comment:
NIT:
you can create the `available_channel_names` in the first for loop (saves another run on the channels list)